snakemake/logging.py

```
@@ -453,324 +612,288 @@
+                    self.last_msg_was_job_info = False
+                formatted_message = self.format(record)
+
+                # Apply color to the formatted message
+                self.stream.write(self.decorate(record, formatted_message))
+                self.stream.write(getattr(self, "terminator", "\n"))
+                self.flush()
+            except BrokenPipeError as e:
+                raise e
+            except (KeyboardInterrupt, SystemExit):
+                pass  # Ignore exceptions for these cases
+            except Exception as e:
```

Ruff

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)